src/server.js

```diff
@@ -22,47 +23,82 @@ const reading = {
   name: "reading philosophy books"
 };
 
+const allSkills = [donutEating, saxophonePlaying, reading];
+
 const userSkill = (skill, regnalNumber) => ({
   skill,
   level: Math.pow(2, regnalNumber)
 });
 
-const getSkills = (name, regnalNumber) => {
-  const skills =
-    name.toLowerCase() === "lisa" ? [saxophonePlaying, reading] : [donutEating];
-
-  return skills.map(skill => userSkill(skill, regnalNumber));
-};
+const userSkills = (skills, regnalNumber) =>
+  skills.map(skill => userSkill(skill, regnalNumber));
 
 const computeRegnalNumber = name =>
-  users.reduce(
-    (count, user) => (user.firstName === name ? count + 1 : count),
+  Object.keys(users).reduce(
+    (count, userId) => (users[userId].firstName === name ? count + 1 : count),
     1
   );
 
-const createUser = ({ firstName, lastName }) => {
+const createUser = ({ firstName, lastName, skills }) => {
   const regnalNumber = computeRegnalNumber(firstName);
+  const id = `user-${userCount}`;
+
+  userCount += 1;
 
   return {
-    id: `user-${users.length}`,
+    id,
     firstName,
     lastName,
     regnalNumber,
-    skills: getSkills(firstName, regnalNumber)
+    skills: userSkills(skills, regnalNumber)
   };
 };
 
-const getUser = id => users.find(user => user.id === id);
+app.get("/users", (req, res) => {
+  const u = Object.keys(users).reduce(
+    (userArray, userId) => [...userArray, users[userId]],
+    []
+  );
+  res.json(u);
+});
+
+app.get("/skills", (req, res) => res.json(allSkills));
+
+app.get("/users/:id", (req, res) => {
+  if (!users[req.params.id]) {
```

Comment:
Might be better to save this into a variable because it's used in 2 places.
```js
const user = users[req.params.id];
```

Reply:
Makes sense - variable created